src/common/http.ts

```diff
@@ -112,3 +153,18 @@ export class TurboHTTPService implements TurboHTTPServiceInterface {
     }
   }
 }
+
+async function toFetchBody(
+  data: ReadableStream<Uint8Array>,
+): Promise<{ body: BodyInit; duplex?: 'half' }> {
+  // Browser ReadableStream
+  if (
+    !navigator.userAgent.includes('Firefox') &&
+    !navigator.userAgent.includes('Safari')
+  ) {
+    return { body: data, duplex: 'half' }; // Chrome / Edge / Opera
+  }
+  // Firefox / Safari fallback: stream → Blob (still streams off-disk)
+  const blob = await new Response(data).blob();
```

Comment:
We're sure this doesn't result in the entire stream getting buffered into memory?

Reply:
not sure if I'm convinced of that either. followed up with o3

<img width="911" height="562" alt="image" src="https://github.com/user-attachments/assets/7eaa0140-e716-4b50-8305-20b38f2c3fea" />

Reply:
seems that it does. going to update this comment. the hard truth here is we just can't do a full streaming until firefox and safari supports it for network requests. but we can allow Chrome users to benefit from the updated streaming 

<img width="999" height="434" alt="image" src="https://github.com/user-attachments/assets/d5f816c7-d244-4729-95d7-d9e94abf12c5" />